### PR TITLE
Added Edit action in Setting Page

### PR DESCRIPTION
**Pull request by hasanmainul**

### CODE REVIEW

File: LocalChat/LocalChat/Controllers/SettingViewController.swift [hunk 1]
```diff
@@ -42,33 +46,56 @@ class SettingViewController: UIViewController, UITableViewDelegate, UITableViewD
     }
```
Comment (Tahia50) [inline LocalChat/LocalChat/Controllers/SettingViewController.swift]: remove the numberOfSections method

File: LocalChat/LocalChat/Controllers/SettingViewController.swift [hunk 2]
```diff
@@ -10,10 +10,13 @@ import UIKit
 
 class SettingViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
     
-    var checked: [Bool] = [false, false, false, false]
+    @IBOutlet weak var editBarButton: UIBarButtonItem!
+    var checked: [Bool] = [false, false, false, false, false, false, false, false]
```
Comment (Tahia50) [inline LocalChat/LocalChat/Controllers/SettingViewController.swift]: u can use repeated value??

File: LocalChat/LocalChat/Controllers/ChatViewController.swift [hunk 3]
```diff
@@ -44,7 +44,8 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
         setSegmentControl()
         
         let botName = userSegmentControl.titleForSegmentAtIndex(userSegmentControl.selectedSegmentIndex) as String!
-        chatUserNameLabel?.text = botName + Constants.userIsChatting
+        chatUserNameLabel?.text = Constants.userIsChatting(botName)
```
Comment (dina28) [inline LocalChat/LocalChat/Controllers/ChatViewController.swift]: what is the meaning of botName??????

File: LocalChat/LocalChat/Controllers/ChatViewController.swift [hunk 4]
```diff
@@ -59,7 +60,7 @@ class ChatViewController: UIViewController, UITextFieldDelegate, UITableViewDele
     
     @IBAction func userSegmentControlAction(sender: AnyObject) {
         let botName = userSegmentControl.titleForSegmentAtIndex(userSegmentControl.selectedSegmentIndex) as String!
-        chatUserNameLabel?.text = botName + Constants.userIsChatting
+        chatUserNameLabel?.text = Constants.userIsChatting(botName)
```
Comment (dina28) [inline LocalChat/LocalChat/Controllers/ChatViewController.swift]: what is the meaning of botName??????

File: LocalChat/LocalChat/Controllers/SettingViewController.swift [hunk 5]
```diff
@@ -37,38 +41,57 @@ class SettingViewController: UIViewController, UITableViewDelegate, UITableViewD
...
     }
     
     func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCellWithIdentifier("settingCell", forIndexPath: indexPath)
-        if !checked[indexPath.row] {
-            cell.accessoryType = .None
-        } else if checked[indexPath.row] {
-            cell.accessoryType = .Checkmark
+        if editButtonStartPress {
+            let cell = tableView.dequeueReusableCellWithIdentifier("settingCell", forIndexPath: indexPath)
```
Comment (dina28) [inline LocalChat/LocalChat/Controllers/SettingViewController.swift]: Don't use static value

File: LocalChat/LocalChat/Controllers/SettingViewController.swift [hunk 6]
```diff
@@ -37,38 +41,57 @@ class SettingViewController: UIViewController, UITableViewDelegate, UITableViewD
...
+            
+            var dict = userList[indexPath.row]
+            let userName = dict[Constants.userManagerDictionary.keyName] as? String
+            let avatarURL = dict[Constants.userManagerDictionary.keyAvatar] as? String
+            cell.textLabel!.text = userName
+            cell.imageView?.image = UIImage(named: avatarURL!)
+            
+            return cell
+        } else {
+            let cell = tableView.dequeueReusableCellWithIdentifier("customSettingCell", forIndexPath: indexPath) as! EditUserCell
```
Comment (dina28) [inline LocalChat/LocalChat/Controllers/SettingViewController.swift]: Don't use static value

File: LocalChat/LocalChat/CustomCells/EditUserCell.swift [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+class EditUserCell: UITableViewCell {
+
+    @IBOutlet weak var deleteButton: UIButton!
+    @IBOutlet weak var editableUserNameTextField: UITextField!
+    @IBOutlet weak var editableUserAvatar: UIImageView!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
```
Comment (dina28) [inline LocalChat/LocalChat/CustomCells/EditUserCell.swift]: remove unnecessary lines

File: LocalChat/LocalChat/CustomCells/EditUserCell.swift [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    override func setSelected(selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
```
Comment (dina28) [inline LocalChat/LocalChat/CustomCells/EditUserCell.swift]: remove unnecessary lines

**Comment by dina28** [comment]
👍

**Pull request merged**